src/test/ui/associated-type-bounds/issue-70292.rs

```diff
@@ -0,0 +1,21 @@
+// check-pass
+
+#![feature(associated_type_bounds)]
+
+fn foo<F>(_: F)
+where
+    F: for<'a> Trait<Output: 'a>,
```

Comment:
This test debug asserts (I think it's a problem of `Binder::dummy` being used instead of `Binder::bind`

Reply:
You mean this still triggers ICE with debug assertion enabled, right? If so, should it be resolved in this PR, or should I leave a comment to follow-up later?

Reply:
CI runs with debug assertions enabled on some targets, so this needs to be fixed in this PR, or the tests can be left out until a future PR.

Reply:
Oh, I didn't remember that, thanks for pointing out and finding the tests failing! 9f7eab4 should address that issue.